src/virt.c

```diff
@@ -578,15 +613,23 @@ lv_read (void)
                     &stats, sizeof stats) != 0)
             continue;
 
+	char *new_path=NULL;
```

Comment:
The path is not always NEW but in case of blockdevice_format_basename == true it is new :).
I do not understand why type_instance is more suitable, but if you think it makes more sense I am ok with it. Fixed as suggested.